pennylane/qnn/iqp.py

```diff
@@ -0,0 +1,388 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule defines methods for estimating the expectations of Pauli-Z operators following an IQP circuit.
+"""
+import numpy as np
+from scipy.sparse import csr_matrix, dok_matrix
+
+has_jax = True
+try:
+    import jax
+    import jax.numpy as jnp
+    from jax.numpy import array
```

Comment:
I think this `array` symbol has been only used once? Guess we can use `jnp.array` there as well